network-api/networkapi/wagtailpages/pagemodels/buyersguide/call_to_action.py

```diff
@@ -0,0 +1,93 @@
+from django.forms.utils import ErrorList
+from networkapi.wagtailpages.pagemodels.customblocks.base_rich_text_options import base_rich_text_options
+
+
+@register_snippet
+class BuyersGuideCallToAction(
+    index.Indexed,
+    TranslatableMixin,
+    LocalizedSnippet,
+    models.Model
+):
+    """
+    Reusable call to action for the buyers guide,
+    features a title and rich text body, with optional image and link.
+    """
+    sticker_image = models.ForeignKey(
+        'wagtailimages.Image',
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        verbose_name='Sticker Image',
+        help_text='Optional image on CTA.',
+    )
+    title = models.CharField(max_length=200)
+    content = RichTextField(features=base_rich_text_options, blank=True)
+    link_label = models.CharField(max_length=2048, blank=True)
+    link_target_url = models.URLField(blank=True)
+    link_target_page = models.ForeignKey(
+        Page,
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name='cta_link_page'
+    )
+
+    panels = [
+        ImageChooserPanel("sticker_image"),
+        FieldPanel('title'),
+        FieldPanel('content'),
+        MultiFieldPanel([
+            FieldPanel('link_label'),
+            FieldPanel('link_target_url'),
+            PageChooserPanel('link_target_page'),
+        ], heading='Call To Action Link'),
+    ]
+
+    translatable_fields = [
+        SynchronizedField('sticker_image'),
+        TranslatableField('title'),
+        TranslatableField('content'),
+        TranslatableField('link_label'),
+    ]
+
+    class Meta(TranslatableMixin.Meta):
+        verbose_name = "Buyers Guide Call To Action"
+        verbose_name_plural = "Buyers Guide Call To Actions"
+
+    def __str__(self):
+        return str(self.title)
+
+    def get_target_url(self):
+        if self.link_target_url:
+            return self.link_target_url
+        else:
+            return self.link_target_page.url
+
+    def clean(self):
+        errors = {}
+        duplicate_link_target_error = ErrorList(['Please select a page OR enter a URL for the link (choose one)'])
+        # If user enters both link URL and link page:
+        if self.link_label and self.link_target_url and self.link_target_page:
+            errors['link_target_url'] = duplicate_link_target_error
+            errors['link_target_page'] = duplicate_link_target_error
+        # If user enters link URL or page but no label:
+        elif self.link_target_page or self.link_target_url and not self.link_label:
+            errors['link_label'] = ErrorList(['Please enter a label for the link'])
+        # If user enters link label but no page or URL to link to:
+        elif not self.link_target_page and not self.link_target_url and self.link_label:
+            errors['link_target_url'] = duplicate_link_target_error
+            errors['link_target_page'] = duplicate_link_target_error
```

Comment:
**Question** Why did you choose this ordering? It seems to me like the first and the last case are similar, but they are separated by a different case. 

Also, have you considered adding parenthesis to group the conditions for readability?

Reply:
Updated ordering and added parenthesis for readability